Tools/peg_generator/pegen/parser_generator.py

```diff
@@ -27,6 +27,11 @@ def visit_NameLeaf(self, node: NameLeaf) -> None:
             # TODO: Add line/col info to (leaf) nodes
             raise GrammarError(f"Dangling reference to rule {node.value!r}")
 
+    def visit_NamedItem(self, node: NameLeaf) -> None:
+        if node.name and node.name.startswith("_"):
+            raise GrammarError(f"Variable name '{node.name}' collides with parser reserved names")
```

Comment:
That's not a great error message. How about "Variable names cannot start with underscore"? (Similar below.)